.github/workflows/ci.yml

```
@@ -1,30 +1,14 @@
 name: CI
 
-on:
-  - pull_request
-  - push
+on: pull_request
```

In the examples we simplified it to just `pull_request`, but in practice there is a benefit to using `push`: it builds a common cache. To prevent duplicate runs you can limit it to just running on `master`:
```suggestion
on:
  pull_request:
  push:
    branches:
      - master
```

---

Thanks for your patience ;-)

---

Sorry misunderstood you, its about discrete units of changes, not how many files your chaning per PR right. Thanks

---

Yes, indeed. Though I think it ok to have multiple commits in the same PR.